src/runtime_src/core/edge/user/aie_sys.h

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef _XCL_AIE_SYS_H_
+#define _XCL_AIE_SYS_H_
+
+#include <string>
+#include <vector>
+#include <fstream>
+
+namespace aie_sys_parser {
+
+static std::fstream sysfs_open_path(const std::string& path, std::string& err,
+    bool write, bool binary)
+{   
+    std::fstream fs;
+    std::ios::openmode mode = write ? std::ios::out : std::ios::in;
+    
+    if (binary) 
+        mode |= std::ios::binary;
+    
+    err.clear();
+    fs.open(path, mode);
+    if (!fs.is_open()) {
+        std::stringstream ss;
+        ss << "Failed to open " << path << " for "
```

Comment:
Please use boost::format.  For example:
```
err = boost::str(boost::format("Failed to open %s for %s %s: (%d) %s") 
                                             % path 
                                             % (binary ? "binary " : "") 
                                             % (write ? "writing" : "reading") 
                                             % errno
                                             % strerror(errno));
```
and then throw the exception:
`throw std::runtime_error(err);`

Reply:
fixed